parley/src/tests/utils/env.rs

```diff
@@ -336,15 +343,37 @@ impl TestEnv {
         let snapshot_path = snapshot_dir().join(&image_name);
         let comparison_path = current_imgs_dir().join(&image_name);
 
+        #[track_caller]
+        fn save_image(image: &Pixmap, path: &PathBuf, max_size: Option<usize>) {
+            let image_data = image.encode_png().unwrap();
+
+            let data = optimize_from_memory(&image_data, &Options::from_preset(5)).unwrap();
+            let saved_len = data.len();
+
+            // Whenever we save a file, we optimise it just in case.
+            // In some cases this is redundant with what kompari already does,
+            // but it's simpler to cover all cases.
+            std::fs::write(path, data).unwrap();
+
+            if let Some(max_size) = max_size {
+                if saved_len > max_size {
+                    panic!(
+                        "New screenshot file ({saved_len} bytes) was larger than the supported file size ({max_size} bytes).",
+                    );
+                }
+            }
```

Comment:
I think we need to make sure that this check also applies to the existing file (when we didn't have this in Vello, we had a case where someone ignored the failure and manually renamed the file and submitted the test anyway). If I'm understanding this correctly, this check in this form currently only happens if the images didn't match in size, right?

Reply:
It only applies to new images, yes.

You think we should check existing images? We currently don't to that in Xilem AFAIK.